src/Generators/DurableTaskSourceGenerator.cs

```diff
@@ -15,7 +16,7 @@ namespace Microsoft.DurableTask.Generators
     /// Generator for DurableTask.
     /// </summary>
     [Generator]
-    public class DurableTaskSourceGenerator : ISourceGenerator
+    public class DurableTaskSourceGenerator : IIncrementalGenerator
```

Comment:
Nit: Unless this is required for the `net6` -> `net8/10` upgrade, I would prefer it in another PR to keep things focused :)

Reply:
it's required